satpy/tests/reader_tests/test_nwcsaf_nc.py

```diff
@@ -16,84 +16,271 @@
+                        " 230 239 248 257 266 276 286 296 307 317 328 340 351 363 375 388 401 414 428"
+                        " 442 456 470 485 501 517 533 550 567 584 602 621 640 660 680 700 721 743 765"
+                        " 788 811 835 860 885 911 938 965 993 1022 1052 1082 1113 1145 1178 1212 1246"
+                        " 1282 1318 1355 1394 1433 1474 1515 1558 1601 1646 1692 1739 1788 1837 1889 "
+                        "1941 1995 2050 2107 2165 2224 2286 2348 2413 2479 2547 2617 2688 2762 2837 "
+                        "2915 2994 3076 3159 3245 3333 3424 3517 3612 3710 3810 3913 4019 4127 4239 "
+                        "4353 4470 4591 4714 4841 4971 5105 5242 5383 5527 5676 5828 5984 6144 6309 "
+                        "6478 6651 6829 7011 7199 7391 7588 7791 7999 8212 8431 8656 8886 9123 9366 "
+                        "9615 9871 10134 10404 10680 10964 11256 11555 11862 12177 12501 12833 13173 "
+                        "13523 13882 14250 14628 15016 15414 15823 16243 16673 17115 17569 18034 18512"
+                        " 19002 19505 20022 20552 21096 21654 22227 22816 23419 24039 24675 25327 "
+                        "25997 26685 27390 28114 28858 29621 30404 31207 32032 32878 33747 34639 35554"
+                        " 36493 37457 38446 39462 40504 41574 42672 43798 44955 46142 47360 48610 "
+                        "49893 51210 52562 53949 55373 56834 58334 59873 61453 63075 64739")
+
+COT_SCALE = 0.01
+COT_OFFSET = 0.0
+
+CRE_ARRAY = np.random.randint(0, 65535, size=(928, 1530), dtype=np.uint16)
+COT_ARRAY = np.random.randint(0, 65535, size=(928, 1530), dtype=np.uint16)
+PAL_ARRAY = np.random.randint(0, 255, size=(250, 3), dtype=np.uint8)
+
+
+@pytest.fixture(scope="session")
+def nwcsaf_geo_ct_filename(tmp_path_factory):
+    """Create a CT file and return the filename."""
+    return create_nwcsaf_geo_ct_file(tmp_path_factory.mktemp("data"))
+
+
+def create_nwcsaf_geo_ct_file(directory):
+    """Create a CT file."""
+    filename = directory / "S_NWC_CT_MSG4_MSG-N-VISIR_20230118T103000Z_PLAX.nc"
+    with h5netcdf.File(filename, mode="w") as nc_file:
+        nc_file.dimensions = dimensions
+        nc_file.attrs.update(global_attrs)
+        var_name = "ct"
+
+        var = nc_file.create_variable(var_name, ("ny", "nx"), np.uint16,
+                                      chunks=(256, 256))
+        var[:] = np.random.randint(0, 255, size=(928, 1530), dtype=np.uint8)
+
+    return filename
+
+
+@pytest.fixture
+def nwcsaf_geo_ct_filehandler(nwcsaf_geo_ct_filename):
+    """Create a CT filehandler."""
+    return NcNWCSAF(nwcsaf_geo_ct_filename, {}, {})
+
+
+@pytest.fixture(scope="session")
+def nwcsaf_pps_cmic_filename(tmp_path_factory):
+    """Create a CMIC file."""
+    filename = tmp_path_factory.mktemp("data") / "S_NWC_CMIC_npp_00000_20230118T1427508Z_20230118T1429150Z.nc"
+    with h5netcdf.File(filename, mode="w") as nc_file:
```

Comment:
Maybe you could create a small function to create the actual file, which could be used when creating the cpp file as well, so we reduce duplication?
```suggestion
    create_pps_cloud_microphysics_file(filename, 'cmic')
```

Reply:
Done!